api/src/main/java/org/pmiops/workbench/db/dao/UserServiceImpl.java

```diff
@@ -210,36 +212,56 @@ private void updateUserAccessTiers(DbUser dbUser, Agent agent) {
     tiersForRemoval.forEach(tier -> accessTierService.removeUserFromTier(dbUser, tier));
   }
 
-  private boolean shouldUserBeRegistered(DbUser user) {
-    boolean dataUseAgreementCompliant = false;
+  public boolean isCompleteAndNotExpired(Timestamp completionTime) {
+    Timestamp expirationTime =
+        new Timestamp(
+            clock.millis()
+                - TimeUnit.MILLISECONDS.convert(
+                    configProvider.get().accessRenewal.expiryDays, TimeUnit.DAYS));
+    if (configProvider.get().access.enableAccessRenewal) {
+      return completionTime != null && expirationTime.before(completionTime);
+    }
+    return true;
```

Comment:
This function should return `false` if `completionTime` is null, with or without the feature flag